### PR TITLE
Move RoundRobin PeerList out of /x/

### PR DESCRIPTION
**Pull request by willhug**
Summary: After #1005 the round robin peer list API
is ready to go 1.0.  We may make changes to the internals of the peer
list, but we will not change the API surface of the Config Spec, and the
New function (with list options).

In the process of this diff I removed the api surface of the
PeerRing because that should have never been public, luckily no one
uses it (according to code sleuthing).

I'm leaving behind a "Deprecated" version of the New and Capacity
functions since those are the only functions that are being used.
Once we update the users, we can remove the deprecated functions.

Test Plan: Tests still pass, wrote a deprecated create function.

Depends on: #1005

### CODE REVIEW

File: peer/x/roundrobin/deprecated.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+	"go.uber.org/yarpc/api/peer"
+	newrr "go.uber.org/yarpc/peer/roundrobin"
+)
+
+// Capacity is DEPRECATED, see "go.uber.org/yarpc/peer/roundrobin#Capacity"
+func Capacity(cap int) newrr.ListOption {
+	return newrr.Capacity(cap)
+}
+
+// New is DEPRECATED, see "go.uber.org/yarpc/peer/roundrobin#New"
```
Comment (abhinav) [inline peer/x/roundrobin/deprecated.go]: "http://"?

File: peer/x/roundrobin/deprecated.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package roundrobin
+
+import (
+	"go.uber.org/yarpc/api/peer"
+	newrr "go.uber.org/yarpc/peer/roundrobin"
```
Comment (abhinav) [inline peer/x/roundrobin/deprecated.go]: aliasing not necessary IMO

File: CHANGELOG.md [hunk 3]
```diff
@@ -13,6 +13,7 @@ v1.9.0-dev (unreleased)
     x/config.
 -   x/roundrobin: Added support for taking peer list updates before and after
     the peer list has been started.
+-   Moved the RoundRobin Peer List out of the /x/ package
```
Comment (abhinav) [inline CHANGELOG.md]: .

File: peer/roundrobin/peerring.go [hunk 4]
```diff
@@ -102,15 +102,15 @@ func (pr *PeerRing) RemoveAll() []peer.Peer {
 }
 
 // All returns a snapshot of all the peers from the ring as a list.
-func (pr *PeerRing) All() []peer.Peer {
+func (pr *peerRing) All() []peer.Peer {
```
Comment (abhinav) [inline peer/roundrobin/peerring.go]: sounds like this function is never called?

File: peer/x/roundrobin/deprecated.go (hunk 2 above)
Comment (willhug) [inline peer/x/roundrobin/deprecated.go]: i find it preferable to conflicting with the package namespace *shrug*

File: peer/roundrobin/peerring.go (hunk 4 above)
Comment (willhug) [inline peer/roundrobin/peerring.go]: used in introspect

**Pull request merged**